File: 9-palindrome-number/9-palindrome-number.js
/**
 * @param {number} x
 * @return {boolean}
 */
const isPalindrome = function(x) {
    if (x < 0) return false;
    
    let div = 1;
    while (x >= div * 10) div *= 10;
    
    while (x) {
        const left = ~~(x / div);
        const right = x % 10;
        if (left !== right) return false;
        x = ~~((x % div) / 10);
        div /= 100;
    }
    
    return true;
};

File: easy/9-palindrome-number/9-palindrome-number.rb
# @param {Integer} x
# @return {Boolean}
def is_palindrome(x)
    return false if x < 0
    
    div = 1
    while x >= div*10 do
       div *= 10 
    end
    
    while x != 0 do
        l = x / div
        r = x % 10
        return false if l != r
        x = (x % div) / 10 
        div /= 100
    end
    
    true
end